### PR TITLE
real governance for PSM

### LINKED ISSUE #5153: Add actual governance to PSM module
## What is the Problem Being Solved?

The PSM module [currently doesn't have](https://github.com/Agoric/agoric-sdk/blob/ba8b3073672f6b80d98b5b72c6538b9962196fd5/packages/run-protocol/src/psm/psm.js#L77) its parameters  managed under governance.

## Description of the Design

Add ParamGovernance.

## Security Considerations

Standard governance legibility concerns.

## Test Plan

Verify that the params can be changed by governance.
Notice that starting up the PSM will become more complex, since a Governor will be required.

### PR DESCRIPTION
**Pull request by turadg**
closes: #5153

## Description

Replaces fake governance on PSM with real.

Some other cleanup along the way. Will do a merge commit.

### Security Considerations

Governable parameters.

### Documentation Considerations

--

### Testing Considerations

Made tests set up governance.

### CODE REVIEW

File: packages/run-protocol/test/psm/test-psm.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ import { makeTracer } from '../../src/makeTracer.js';
 import { unsafeMakeBundleCache } from '../bundleTool.js';
 import { setUpZoeForTest } from '../supports.js';
 
+/** @type {import('ava').TestInterface<Awaited<ReturnType<makeTestContext>>>} */
+// @ts-expect-error cast
+const test = anyTest;
```
Comment (dckc) [inline packages/run-protocol/test/psm/test-psm.js:25]: nice :)

File: packages/run-protocol/src/psm/psm.js [hunk 2]
```diff
@@ -67,31 +66,20 @@ export const start = async (zcf, privateArgs) => {
       anchorPerStable.denominator.brand === stableBrand,
     X`Ratio ${anchorPerStable} is not consistent with brands ${anchorBrand} and ${stableBrand}`,
   );
+  const bpRatio = bp => makeRatio(bp, stableBrand, BASIS_POINTS);
```
Comment (dckc) [inline packages/run-protocol/src/psm/psm.js]: oops; I forgot to tell you: @dtribble said basis points are too coarse; he wants the governance parameters to be ratios.

File: packages/run-protocol/src/psm/psm.js [hunk 3]
```diff
@@ -103,13 +91,13 @@ export const start = async (zcf, privateArgs) => {
...
     assert(
-      AmountMath.isGTE(gov.getMintLimit(), anchorAfterTrade),
+      AmountMath.isGTE(params.getMintLimit(), anchorAfterTrade),
       X`Request would exceed mint limit`,
     );
   };
 
   const giveStable = (seat, given, wanted = emptyAnchor) => {
-    const fee = ceilMultiplyBy(given, gov.getGiveStableRate());
+    const fee = ceilMultiplyBy(given, bpRatio(params.getGiveStableFeeBP()));
```
Comment (dckc) [inline packages/run-protocol/src/psm/psm.js]: likewise...

File: packages/run-protocol/src/psm/psm.js [hunk 4]
```diff
@@ -127,7 +115,7 @@ export const start = async (zcf, privateArgs) => {
   const wantStable = (seat, given, wanted = emptyStable) => {
     assertUnderLimit(given);
     const asStable = floorDivideBy(given, anchorPerStable);
-    const fee = ceilMultiplyBy(asStable, gov.getWantStableRate());
+    const fee = ceilMultiplyBy(asStable, bpRatio(params.getWantStableFeeBP()));
```
Comment (dckc) [inline packages/run-protocol/src/psm/psm.js]: and again

**Review by dckc — CHANGES_REQUESTED** [review]
please change the NAT basis points parameters to ratios; sorry I forgot to tell you about the additional requirement.

Strictly speaking it could be a separate PR, but it seems expedient to include it here.

File: packages/run-protocol/src/psm/psm.js (hunk 2 above)
Comment (turadg) [reply packages/run-protocol/src/psm/psm.js]: We don't have a brand-less ratio. We could do a pair of nats to make one.

Is there a grain size that would be sufficient, like 10^-12?

Comment (dckc) [reply packages/run-protocol/src/psm/psm.js]: I don't see a need for brand-less ratios. The caller would have to supply the right brands. Am I confused?

Comment (turadg) [reply packages/run-protocol/src/psm/psm.js]: Cool. I was confusing this with another conversation in RUN meeting.

Done.

**Review by dckc — APPROVED** [review]
bingo! thanks.

**Comment by dckc** [comment]
oops; this is going to break the psm bootstrap code in ways that I'm not sure our tests cover. Oh well. that bootstrap code is kinda iffy anyway.

File: packages/run-protocol/src/psm/psm.js [hunk 5]
```diff
@@ -74,42 +70,38 @@ export const start = async (zcf, privateArgs) => {
...
-      AmountMath.isGTE(gov.getMintLimit(), anchorAfterTrade),
+      AmountMath.isGTE(params.getMintLimit(), anchorAfterTrade),
       X`Request would exceed mint limit`,
     );
   };
 
+  /**
+   * @param {ZCFSeat} seat
+   * @param {Amount<'nat'>} given
+   * @param {Amount<'nat'>} wanted
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/psm/psm.js]: It looks like `wanted` is optional. Is this the right syntax for that?
```suggestion
   * @param {Amount<'nat'>} [wanted]
```

File: packages/run-protocol/src/psm/psm.js [hunk 6]
```diff
@@ -124,10 +116,15 @@ export const start = async (zcf, privateArgs) => {
     stableMint.burnLosses({ Stable: afterFee }, stage);
   };
 
+  /**
+   * @param {ZCFSeat} seat
+   * @param {Amount<'nat'>} given
+   * @param {Amount<'nat'>} wanted
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/psm/psm.js]: optional

**Review by Chris-Hibbert — APPROVED** [review]

**Comment by dckc** [comment]
@turadg did you see @Chris-Hibbert 's optional comments? turning off automerge for a sec...

File: packages/run-protocol/src/psm/psm.js (hunk 5 above)
Comment (turadg) [reply packages/run-protocol/src/psm/psm.js]: Good catch. Turns out [Typescript infers optional from a default value](https://www.typescriptlang.org/play?filetype=js#code/PQKhFgCgAIWgBADgQwE7ILbQN4GcAuqAlgHYDmAvtANoGoC6UswUAxgPYkHTuL5GdkAG2gBeaAAo6ASjEA+HE2h0AdBxKtk+CdIDcUCvshRe-QUJ1GooCDDhI0mHHVKVlhJiBaR13ACYApgBmyACuQvgBfmKSdKIA5AAWRPGyogrYSqrqmtp6BlaQgSHhkX6WUEA), but this is correct JSDoc.

**Pull request merged**